File: ObjC/CircularSegmentedControl/CircularSegmentedControl/Segment.m
//
//  Segment.m
//  CircularSegmentedControl
//
//  Created by Don Mag on 12/9/24.
//

#import "Segment.h"

@implementation Segment

- (instancetype)init {
	self = [super init];
	if (self) {
		_title = @"A";
		_startAngleInDegrees = 0.0;
		_endAngleInDegrees = 0.0;
		_path = [[UIBezierPath alloc] init];
	}
	return self;
}

- (double)midAngleInDegrees {
	return (self.startAngleInDegrees + self.endAngleInDegrees) * 0.5;
}

@end

File: Swift/CircularSegmentedControl/CircularSegmentedControl/Segment.swift
//
//  Segment.swift
//  CircularSegmentedControl
//
//  Created by Don Mag on 12/11/24.
//

import UIKit

class Segment: NSObject {
	var title: String = "A"
	var startAngleInDegrees: Double = 0.0
	var endAngleInDegrees: Double = 0.0
	var path: UIBezierPath = UIBezierPath()
}
